missioncontrol/etl/measure.py

```diff
@@ -0,0 +1,90 @@
+import datetime
+import logging
+from distutils.version import LooseVersion
+
+from django.core.cache import cache
+
+from missioncontrol.celery import celery
+from missioncontrol.settings import DATA_EXPIRY_INTERVAL
+from .presto import raw_query
+from .schema import (CHANNELS, TELEMETRY_PLATFORM_MAPPING, get_measure_cache_key)
+from .versions import get_firefox_versions
+
+logger = logging.getLogger(__name__)
+
+
+@celery.task
+def update_measure(platform_name, channel_name, measure_name):
+    logger.info('Updating measure: %s %s %s', channel_name, platform_name,
+                measure_name)
+    channel = CHANNELS[channel_name]
+    cache_key = get_measure_cache_key(platform_name, channel_name, measure_name)
+
+    # data is stored in this format:
+    # {
+    #   'buildid': {
+    #     'version': 'XXX',
+    #     'data': [[<timestamp>, value, usage hours], [<timestamp>, value, usage hours], ...]
+    #   },
+    #   'buildid2': {
+    #       ...
+    #   },
+    #  ...
+    # }
+    min_timestamp = datetime.datetime.utcnow() - DATA_EXPIRY_INTERVAL
+    min_buildid_timestamp = min_timestamp - channel['update_interval']
+    data = cache.get(cache_key, {})
+    if data:
+        min_timestamp = max([min_timestamp] +
+                            [existing_build_data['data'][-1][0]
+                             for existing_build_data in data.values()])
+
+    # also place a restriction on version (to avoid fetching data
+    # for bogus versions)
+    versions = get_firefox_versions()
+    current_version = versions[channel_name]
+    if channel == 'esr':
+        min_version = str(LooseVersion(versions[channel_name]).version[0] - 7)
+    else:
+        min_version = str(LooseVersion(versions[channel_name]).version[0] - 1)
+
+    raw_sql = '''
```

Comment:
It looks like it is not possible to use parameters to specify column names -- if I do, i get this:

```txt
sqlalchemy.exc.OperationalError: (pyathena.error.OperationalError) SYNTAX_ERROR: line 1:41: Unexpected parameters (varchar(12)) for function sum. Expected: sum(double) , sum(real) , sum(bigint) , sum(decimal(p,s))  [SQL: "select window_start, build_id, version, sum(%(measure)s) as measure_count, sum(usage_hours) as usage_hours        from error_aggregates where application='Firefox' and        version >= %(min_version)s and version <= %(current_version)s and        build_id > %(min_build_id)s and        os_name=%(os_name)s and        channel=%(channel_name)s and        window_start > timestamp %(min_timestamp)s        group by (window_start, build_id, version)"] [parameters: {'min_build_id': '20170523', 'min_version': '54', 'channel_name': 'release', 'min_timestamp': '2017-08-17 18:00:00', 'measure': 'main_crashes', 'current_version': '55.0.2', 'os_name': 'Windows_NT'}]
```

However, it looks like I can do this for the rest of the parameters, so I will do so.

BTW, I'd like to have some kind of easy entrypoint into redash from the measure detail view in the ui, in case people care to look into a particular combination of dimensions in more detail -- this probably implies storing these presto/sql statements somewhere. But we can get to that later.